src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CppCodegenNodeFactory.cs

```diff
@@ -38,5 +44,33 @@ protected override ISymbolNode CreateReadyToRunHelperNode(Tuple<ReadyToRunHelper
             // TODO: this is wrong: this returns an assembly stub node
             return new ReadyToRunHelperNode(helperCall.Item1, helperCall.Item2);
         }
+
+        private void AddWellKnownType(WellKnownType wellKnownType, DependencyAnalysisFramework.DependencyAnalyzerBase<NodeFactory> graph)
+        {
+            var type = TypeSystemContext.GetWellKnownType(wellKnownType);
+            var typeNode = ConstructedTypeSymbol(type);
+            graph.AddRoot(typeNode, "Enables CPP codegen");
+        }
+        private void AddWellKnownTypes(DependencyAnalysisFramework.DependencyAnalyzerBase<NodeFactory> graph)
+        {
+
+            AddWellKnownType(WellKnownType.Void, graph);
```

Comment:
`_wellKnownTypeNodes ` was a workaround for not rooting well-known types before. All of the well-known types are not always constructed, so their VTables isn't emitted  (which we need for the C++ to compile). Instead of manually rooting another approach might be manually requesting a constructed type node?